docs/src/man/basics.md

```diff
@@ -0,0 +1,2002 @@
+     │ Int64  String  Int64
+─────┼──────────────────────
+   1 │    80  male        2
+   2 │    85  female      2
+   3 │    98  male        1
+   4 │    95  male        2
+   5 │    78  male        2
+   6 │    89  male        1
+
+julia> df1.Age === val # no copy is performed
+true
+
+julia> df1[1:3, :Job] = [2, 3, 2] # set value of `:semester` in row `1:3` to values `[2, 4, 6]` *in-place*
+3-element Vector{Int64}:
+ 2
+ 3
+ 2
+
+julia> df1
+6×3 DataFrame
+ Row │ Age    Sex     Job
+     │ Int64  String  Int64
+─────┼──────────────────────
+   1 │    80  male        2
+   2 │    85  female      3
+   3 │    98  male        2
+   4 │    95  male        2
+   5 │    78  male        2
+   6 │    89  male        1
+
+julia> df1[!, :Sex] = ["male", "female", "female", "transgender", "female", "male"] # replaces `:Sex` without copying
+6-element Vector{String}:
+ "male"
+ "female"
+ "female"
+ "transgender"
+ "female"
+ "male"
+
+julia> df1
+6×3 DataFrame
+ Row │ Age    Sex          Job
+     │ Int64  String       Int64
+─────┼───────────────────────────
+   1 │    80  male             2
+   2 │    85  female           3
+   3 │    98  female           2
+   4 │    95  transgender      2
+   5 │    78  female           2
+   6 │    89  male             1
+
+julia> df1[3, 1:3] = [78, "male", 4] # set row `3` of cols `1:3(:Age, :Sex, :Job)` in-place
+3-element Vector{Any}:
+ 78
+   "male"
+  4
+
+julia> df1
+6×3 DataFrame
+ Row │ Age    Sex          Job
+     │ Int64  String       Int64
+─────┼───────────────────────────
+   1 │    80  male             2
+   2 │    85  female           3
+   3 │    78  male             4
+   4 │    95  transgender      2
+   5 │    78  female           2
+   6 │    89  male             1
+```
+
+Now, lets talk about *DataFrameRow*: 
+
+```jldoctest dataframe
+julia> dfr = df1[2, :] # passed `:`(a column) in a parent DataFrame to get the *DataFrameRow*
+DataFrameRow
+ Row │ Age    Sex     Job
+     │ Int64  String  Int64
+─────┼──────────────────────
+   2 │    85  female      3
+
+julia> dfr
+DataFrameRow
+ Row │ Age    Sex     Job
+     │ Int64  String  Int64
+─────┼──────────────────────
+   2 │    85  female      3
+
+julia> dfr.Age = 98 # set value of col `:Age` in row `2` to `98` in-place
+98
+
+julia> dfr
+DataFrameRow
+ Row │ Age    Sex     Job
+     │ Int64  String  Int64
+─────┼──────────────────────
+   2 │    98  female      3
+
+julia> dfr[2:3] = ["male", 2] # set values of entries in columns `:Sex` and `:Job` 
+2-element Vector{Any}:
+  "male"
+ 2
+
+julia> dfr
+DataFrameRow
+ Row │ Age    Sex     Job
+     │ Int64  String  Int64
+─────┼──────────────────────
+   2 │    98  male        2
+```
+
+As mentioned in the title that we will discuss `setindex!` for *DataFrame*, *SubDataFrame*, *DataFrameRow*. So here
+we are talking about *SubDataFrame* and to create a *SubDataFrame* we need of `view` function. To know more about 
+`view` please have a look to **Views** section of this tutorial. 
```

Comment:
please add cross-link

Reply:
Rather than moving the `views` section, we can add a cross-link or just move `views` section here not to that place where you have mentioned previously.

Reply:
I prefer to have `view` section after `getindex` and before `setindex!`